spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -303,13 +304,13 @@ public RetryTemplate buildRetryTemplate(ConsumerProperties properties) {
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
 		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
+
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
-						: MimeTypeUtils.TEXT_PLAIN;
+				return originalContentType.toLowerCase().contains(MimeTypeUtils.APPLICATION_JSON.getSubtype()) ? MimeType.valueOf(originalContentType) : MimeTypeUtils.TEXT_PLAIN;
```

Comment:
@ilayaperumalg I don't understand why we would want to keep this behavior. It looks like a bad smell to me having this:
`ContentType: application/json` `OriginalContentType: application/vnd.type.version+json` I've never got my head around this.

If can have only `ContentType` why do I want to enforce originalContentType? And it's very awkward that we have only `ContentType` when it's JSON payload, and still we have `originalContentType` for anything else. 

I think there's a big gap to be improved in ContentType handling on SCSt. If we believe that this should wait for a bigger refactor, I'm ok, but I don't see any reason why forcing an extra header in this case.